tech-specs/src/explore/design/ibc.md

```diff
@@ -0,0 +1,408 @@
+
+                    // Assert that the packet sequence is the next sequence that the channel expects
+
+                    // Assert that the timeout height and timestamp have not passed on the destination ledger
+
+                    // Assert that the commitment has been stored
+                }
+
+                "nextSequenceRecv" => {
+                    // "RecvPacket"
+                    let packet = get_packet(key, data_from_tx)?;
+                    let proofs = get_proofs(key, data_from_tx)?;
+
+                    // Assert that the packet metadata matches the channel and connection information
+
+                    // Assert that the connection and channel are open
+
+                    // Assert that the packet sequence is the next sequence that the channel expects (Ordered channel)
+
+                    // Assert that the timeout height and timestamp have not passed on the destination ledger
+
+                    // Assert that the receipt and acknowledgement have been stored
+
+                    // Verify the proofs that the counterpart ledger has stored the commitment
+                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_packet_recv_proofs()`
+                }
+
+                "nextSequenceAck" => {
+                    // "Acknowledgement"
+                    let packet = get_packet(key, data_from_tx)?;
+                    let proofs = get_proofs(key, data_from_tx)?;
+
+                    // Assert that the packet metadata matches the channel and connection information
+
+                    // Assert that the connection and channel are open
+
+                    // Assert that the packet sequence is the next sequence that the channel expects (Ordered channel)
+
+                    // Assert that the commitment has been deleted
+
+                    // Verify that the packet was actually sent on this channel
+                    //   - Get the stored commitment and compare it with a commitment made from the packet
+
+                    // Verify the proofs to check the acknowledgement has been written on the counterpart ledger
+                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_packet_acknowledgement_proofs()`
+                }
+
+                "commitments" => {
+                    let packet = get_packet(key, data_from_tx)?;
+                    // check if the commitment is deleted
+                    match check_commitment_state(key) {
+                        StateChange::Deleted => {
+                            // Assert that the packet was actually sent on this channel
+                            //   - Get the stored commitment and compare it with a commitment made from the packet
+
+                            // Check the channel state change
+                            match check_channel_state(channel_id) {
+                                StateChange::Closed => {
+                                    // "Timeout"
```

Comment:
is there a function in `ibc-rs` we can call here?

Reply:
In "Timeout" case. `ChannelReader`'s functions and [`verify_next_sequence_recv()`](https://github.com/informalsystems/ibc-rs/blob/5cf3b6790c45539c5aaadeef6e1af1f51a5f437f/modules/src/ics04_channel/handler/verify.rs#L134) or [`verify_packet_receipt_absence()`](https://github.com/informalsystems/ibc-rs/blob/5cf3b6790c45539c5aaadeef6e1af1f51a5f437f/modules/src/ics04_channel/handler/verify.rs#L165) are called.